Mage.Sets/src/mage/cards/g/GyrudaDoomOfDepths.java

```diff
@@ -102,15 +100,13 @@ public boolean apply(Game game, Ability source) {
             return false;
         }
         Cards cards = new CardsImpl();
-        game.getState()
-                .getPlayersInRange(source.getControllerId(), game)
-                .stream()
-                .map(game::getPlayer)
-                .map(Player::getLibrary)
-                .map(library -> library.getTopCards(game, 4))
-                .flatMap(Collection::stream)
-                .forEach(cards::add);
-        controller.moveCards(cards, Zone.GRAVEYARD, source, game);
+        for (UUID playerId : game.getState().getPlayersInRange(source.getControllerId(), game)) {
+            Player player = game.getPlayer(playerId);
+            if (player == null) {
+                continue;
+            }
+            cards.addAll(player.millCards(4, source, game));
+        }
         cards.removeIf(cardId -> game.getState().getZone(cardId) != Zone.GRAVEYARD
                 && game.getState().getZone(cardId) != Zone.EXILED);
```

Comment:
Some mill cards checks Zone.GRAVEYARD, but that card checks Zone.EXILED too. But card's text don't contains about exile. So what's correct?

Reply:
It checks whatever zone it went to unless the card specifically mentions the graveyard